### ISSUE TITLE
Traffic Sign Recognition

### ISSUE THREAD

**Issue by keshripritesh** [PRIYANSHU2026/DataSentience-AIML#49]
---

### 📌 Issue: Add Project - *Traffic Sign Recognition*

**Project Type**: Deep Learning / Computer Vision
**Level**: Intermediate
**Technology Stack**: Python, TensorFlow, Keras, Pandas, Scikit-learn, PIL

**Status**: To be reviewed ✅

---

### 🚧 Project Title

**Traffic Sign Recognition using Deep Neural Network**

---

### 🔍 Abstract / Problem Statement

With the increasing reliance on autonomous vehicles and AI-assisted driving systems, accurate recognition of traffic signs is critical for road safety and compliance. The aim of this project is to develop a deep learning-based Traffic Sign Recognition system that can identify and classify various traffic signs using image data. This system can serve as a fundamental model for driver-assistance tools and smart city integrations.

---

### 🧠 Project Description

This project leverages deep neural networks (CNNs) built using Keras and TensorFlow to classify over 40 types of German traffic signs. Trained on a publicly available Kaggle dataset, the model achieves over 95% accuracy in recognizing and predicting traffic sign classes.

It includes a GUI interface to test model predictions in real-time using uploaded images.

---

### 🧩 Key Features

* Over **50,000** training images and **43** traffic sign classes
* Deep Neural Network architecture with high accuracy
* User-friendly **Tkinter GUI** for real-time image testing
* Image pre-processing and data augmentation implemented
* End-to-end pipeline: Load → Train → Evaluate → Predict

---

### 📁 Repository Structure

```
Traffic-Sign-Recognition/
│
├── gui.py               # GUI interface using Tkinter
├── train.py             # Model training script
├── model.h5             # Pre-trained Keras model
├── utils.py             # Image processing utilities
├── Images/
│   └── Demo.jpg         # GUI demo screenshot
└── README.md            # Project documentation
```

---

### 📦 Dependencies

Install with:

```bash
pip install pandas keras tensorflow matplotlib pillow sklearn
```

---

### 📸 Demo

![Demo](Images/Demo.jpg)

---

### 🧪 Dataset

* **Name**: GTSRB - German Traffic Sign Recognition Benchmark
* **Source**: [[Kaggle Dataset](https://www.kaggle.com/datasets/meowmeowmeowmeowmeow/gtsrb-german-traffic-sign)](https://www.kaggle.com/datasets/meowmeowmeowmeowmeow/gtsrb-german-traffic-sign)
* **Description**: Contains labeled images of German traffic signs used for classification.

---

### 💡 Future Scope

* Extend to real-time video feed analysis (OpenCV integration)
* Deploy as a web app using Streamlit or Flask
* Support for multilingual signs
* Model compression for edge devices (e.g., Raspberry Pi)

---

### 🤝 Contribution Guidelines

* Add unit tests for `utils.py` functions
* Optimize CNN layers for better accuracy or faster inference
* Create a Dockerfile for easy containerized deployment
* Improve GUI with drag-and-drop functionality

---

### 🏁 Final Deliverables

* Cleaned codebase with well-commented scripts
* README with installation and usage instructions
* Pre-trained model weights
* Demo GIF / Screenshot
* GUI script for testing

---

**Comment by GauravKarakoti** [comment]
Can i work on this?

**Comment by PRIYANSHU2026** [comment]
yes

**Comment by GauravKarakoti** [comment]
> PR merged.

Please check Discord
